### :x: 3 Tests Failed:
| Tests completed | Failed | Passed | Skipped |
|---|---|---|---|
| 578 | 3 | 575 | 1 |
<details><summary>View the top 3 failed test(s) by shortest run time</summary>

> <pre><code class="language-python">tests.test_integration::test_sensor_metadata_in_redis</code></pre>
> <details><summary>Stack Traces | 3.45s run time</summary>
> 
> > <pre><code class="language-python">client = &lt;eigsep_observing.testing.client.DummyPandaClient object at 0x7f41f8da2e90&gt;
> > transport = &lt;eigsep_redis.testing.transport.DummyTransport object at 0x7f41f8012dd0&gt;
> > 
> >     def test_sensor_metadata_in_redis(client, transport):
> >         &quot;&quot;&quot;Emulators generate status that flows through redis_handler to Redis.&quot;&quot;&quot;
> >         time.sleep(0.5)
> >     
> >         metadata = MetadataSnapshotReader(transport).get()
> >     
> >         expected_sensors = {
> >             &quot;tempctrl&quot;,
> >             &quot;potmon&quot;,
> >             &quot;imu_el&quot;,
> >             &quot;imu_az&quot;,
> >             &quot;lidar&quot;,
> >             &quot;rfswitch&quot;,
> >         }
> >         for sensor in expected_sensors:
> > &gt;           assert sensor in metadata, (
> >                 f&quot;Expected sensor &#x27;{sensor}&#x27; in metadata, &quot;
> >                 f&quot;got keys: {list(metadata.keys())}&quot;
> >             )
> > E           AssertionError: Expected sensor &#x27;tempctrl&#x27; in metadata, got keys: [&#x27;imu_el&#x27;, &#x27;imu_el_ts&#x27;, &#x27;potmon&#x27;, &#x27;potmon_ts&#x27;, &#x27;imu_az&#x27;, &#x27;imu_az_ts&#x27;, &#x27;lidar&#x27;, &#x27;lidar_ts&#x27;, &#x27;rfswitch&#x27;, &#x27;rfswitch_ts&#x27;, &#x27;tempctrl_lna&#x27;, &#x27;tempctrl_lna_ts&#x27;, &#x27;tempctrl_load&#x27;, &#x27;tempctrl_load_ts&#x27;, &#x27;motor&#x27;, &#x27;motor_ts&#x27;]
> > E           assert &#x27;tempctrl&#x27; in {&#x27;imu_az&#x27;: {&#x27;accel_x&#x27;: 0.08648045496880578, &#x27;accel_y&#x27;: -0.0025608474603536495, &#x27;accel_z&#x27;: 9.8094966198456, &#x27;app_id&#x27;: 6...ccel_y&#x27;: 0.00014807117186299887, &#x27;accel_z&#x27;: 9.809664328932513, &#x27;app_id&#x27;: 3, ...}, &#x27;imu_el_ts&#x27;: 1778892801.0011325, ...}
> > 
> > tests/test_integration.py:72: AssertionError</code></pre>
> 
> </details>


> <pre><code class="language-python">tests.test_integration::test_metadata_has_expected_fields</code></pre>
> <details><summary>Stack Traces | 3.91s run time</summary>
> 
> > <pre><code class="language-python">client = &lt;eigsep_observing.testing.client.DummyPandaClient object at 0x7f41f8da1f90&gt;
> > transport = &lt;eigsep_redis.testing.transport.DummyTransport object at 0x7f41f80130d0&gt;
> > 
> >     def test_metadata_has_expected_fields(client, transport):
> >         &quot;&quot;&quot;Verify that metadata values contain the expected sensor fields.&quot;&quot;&quot;
> >         time.sleep(0.5)
> >     
> >         metadata = MetadataSnapshotReader(transport).get()
> >     
> >         # Check tempctrl LNA/LOAD channels
> >         tempctrl = metadata.get(&quot;tempctrl&quot;, {})
> > &gt;       assert &quot;LNA_T_now&quot; in tempctrl
> > E       AssertionError: assert &#x27;LNA_T_now&#x27; in {}
> > 
> > tests/test_integration.py:86: AssertionError</code></pre>
> 
> </details>


> <pre><code class="language-python">tests.test_tempctrl_client::test_get_status_returns_snapshot_or_none</code></pre>
> <details><summary>Stack Traces | 5.45s run time</summary>
> 
> > <pre><code class="language-python">client = &lt;eigsep_observing.testing.client.DummyPandaClient object at 0x7fdce1e94d70&gt;
> > 
> >     def test_get_status_returns_snapshot_or_none(client):
> >         &quot;&quot;&quot;get_status returns a dict with the full schema, or None before
> >         the pico has published. Fire a command first to ensure at least
> >         one publish has happened.&quot;&quot;&quot;
> >         tc = TempCtrlClient(client.transport, settings=SETTINGS)
> >         tc.apply_settings()
> > &gt;       assert _wait_until(lambda: tc.get_status() is not None)
> > E       assert False
> > E        +  where False = _wait_until(&lt;function test_get_status_returns_snapshot_or_none.&lt;locals&gt;.&lt;lambda&gt; at 0x7fdce0b71850&gt;)
> > 
> > tests/test_tempctrl_client.py:169: AssertionError</code></pre>
> 
> </details>

</details>

To view more test analytics, go to the [Test Analytics Dashboard](https://app.codecov.io/gh/EIGSEP/eigsep_observing/tests/feat%2Flive-status-health-tiles)
<sub>📋 Got 3 mins? [Take this short survey](https://forms.gle/22i53Qa1CySZjA6c7) to help us improve Test Analytics.</sub>

--- a/scripts/live_status.py
+++ b/scripts/live_status.py
@@ -83,6 +83,18 @@ def main(argv=None):
     )
     parser.add_argument("--panda-port", type=int, default=6379)
     parser.add_argument(
+        "--snap-fpga-host",
+        default=None,
+        help=(
+            "Override for the SNAP FPGA IP used by the reachability "
+            "probe. Default: read ``snap_ip`` from the corr config "
+            "in Redis once available (10.10.10.12 in the bundled "
+            "corr_config.yaml). Pass this only when corr config has "
+            "not yet been published and you want the SNAP FPGA tile "
+            "to still light up."
+        ),
+    )
+    parser.add_argument(
         "--obs-config",
         default=None,
         help=(
@@ -130,6 +142,7 @@ def main(argv=None):
         transport_panda=transport_panda,
         obs_cfg=obs_cfg,
         thresholds=thresholds,
+        snap_fpga_host_override=args.snap_fpga_host,
     )
 
     app = create_app(aggregator)

--- a/src/eigsep_observing/live_status/aggregator.py
+++ b/src/eigsep_observing/live_status/aggregator.py
@@ -60,6 +60,7 @@ from ..snap_reinit import read as read_snap_reinit
 from ..utils import calc_freqs_dfreq
 from ..vna import VnaReader
 from .signals import SIGNAL_REGISTRY
+from .snap_probe import probe_snap_fpga
 from .thresholds import Thresholds
 from .vna_calibration import VnaCache
 
@@ -197,6 +198,12 @@ class StateSnapshot:
     snap_error: Optional[str] = None
     panda_error: Optional[str] = None
 
+    # SNAP FPGA reachability (independent of corr stream). Probed via
+    # TCP to the katcp port only when corr is stale — corr flow is a
+    # stronger signal and makes the probe redundant.
+    snap_fpga_reachable: Optional[bool] = None
+    snap_fpga_last_probe_unix: Optional[float] = None
+
 
 def corr_observing_timeout_s(state: StateSnapshot) -> float:
     """Infer a reasonable observing-idle timeout from corr cadence.
@@ -265,6 +272,8 @@ class LiveStatusAggregator:
         snap_tick_s: float = 0.5,
         panda_tick_s: float = 0.5,
         read_timeout_s: float = 0.2,
+        snap_fpga_probe_interval_s: float = 5.0,
+        snap_fpga_host: Optional[str] = None,
         stop_event: Optional[threading.Event] = None,
     ):
         if read_timeout_s <= 0:
@@ -278,6 +287,8 @@ class LiveStatusAggregator:
         self._snap_tick_s = snap_tick_s
         self._panda_tick_s = panda_tick_s
         self._read_timeout_s = read_timeout_s
+        self._snap_fpga_probe_interval_s = snap_fpga_probe_interval_s
+        self._snap_fpga_host_override = snap_fpga_host
         self._stop_event = stop_event or threading.Event()
 
         # SNAP-side surfaces.
@@ -402,6 +413,8 @@ class LiveStatusAggregator:
                     if s.last_rfnon_pairs is not None
                     else None
                 ),
+                snap_fpga_reachable=s.snap_fpga_reachable,
+                snap_fpga_last_probe_unix=s.snap_fpga_last_probe_unix,
             )
 
     # -- SNAP drain loop -------------------------------------------
@@ -420,6 +433,12 @@ class LiveStatusAggregator:
         now = time.time()
         errors: list[str] = []
         any_ok = False
+        # Snapshot the fields the probe needs before the main lock
+        # block may overwrite them from Redis.  This lets tests pre-set
+        # state fields and have the probe see them deterministically.
+        with self._lock:
+            _pre_corr_last_unix = self.state.corr_last_unix
+            _pre_corr_cfg = self.state.corr_config
 
         # Corr config + header (pointer-free gets). ``ValueError`` from
         # the store means "header/config not in Redis yet" — that's a
@@ -547,6 +566,59 @@ class LiveStatusAggregator:
 
             if reinit_h is not None:
                 s.snap_reinit = reinit_h
+
+        self._maybe_probe_snap_fpga(now, _pre_corr_last_unix, _pre_corr_cfg)
+
+    def _maybe_probe_snap_fpga(
+        self,
+        now: float,
+        corr_unix: Optional[float],
+        corr_cfg: Optional[dict],
+    ) -> None:
+        """Probe the SNAP FPGA if corr is stale + interval elapsed.
+
+        Parameters are snapshotted by ``_snap_tick`` *before* the main
+        lock block so test pre-sets survive the Redis read. In
+        production the values are from the previous tick, which is
+        fine — the probe is rate-limited to once per
+        ``snap_fpga_probe_interval_s`` anyway.
+
+        Skips the probe entirely when ``corr_unix`` is fresh —
+        the corr stream itself is a stronger liveness signal. Also
+        skips when no ``snap_ip`` is configured anywhere (corr config
+        not yet read AND no CLI override), leaving
+        ``snap_fpga_reachable`` at ``None`` so the tile shows
+        "unknown" rather than a false negative.
+        """
+        with self._lock:
+            last_probe = self.state.snap_fpga_last_probe_unix
+            timeout_s = corr_observing_timeout_s(self.state)
+
+        # Don't probe — corr stream is proving liveness.
+        if corr_unix is not None:
+            if (now - corr_unix) < timeout_s:
+                return
+
+        # Rate-limit the probe.
+        if (
+            last_probe is not None
+            and (now - last_probe) < self._snap_fpga_probe_interval_s
+        ):
+            return
+
+        # Pick the host: corr_config first, CLI override second.
+        snap_ip = None
+        if corr_cfg is not None:
+            snap_ip = corr_cfg.get("snap_ip")
+        if not snap_ip:
+            snap_ip = self._snap_fpga_host_override
+        if not snap_ip:
+            return  # state.snap_fpga_reachable stays None → "unknown"
+
+        result = probe_snap_fpga(snap_ip)
+        with self._lock:
+            self.state.snap_fpga_reachable = result
+            self.state.snap_fpga_last_probe_unix = now
 
     def _read_corr(self) -> tuple[Optional[tuple], bool]:
         """Drain the corr stream to the latest entry; reshape the tail.

--- a/src/eigsep_observing/live_status/app.py
+++ b/src/eigsep_observing/live_status/app.py
@@ -536,6 +536,19 @@ def _health_payload(state: StateSnapshot, now: float) -> dict:
     run_age_s = None
     if state.run_started_at_unix is not None:
         run_age_s = max(0.0, now - state.run_started_at_unix)
+    # Tier the SNAP FPGA tile: corr stream beats probe, probe beats
+    # silence. ``observing_inferred`` and the probe both flag SNAP
+    # liveness; this layer picks the strongest available evidence so
+    # the operator sees "live" during normal observing and only sees
+    # the active-probe result when the observe loop isn't running.
+    if observing_inferred:
+        snap_fpga_state = "live"
+    elif state.snap_fpga_reachable is True:
+        snap_fpga_state = "reachable"
+    elif state.snap_fpga_reachable is False:
+        snap_fpga_state = "unreachable"
+    else:
+        snap_fpga_state = "unknown"
     return {
         "snap_connected": state.snap_connected,
         "panda_connected": state.panda_connected,
@@ -547,6 +560,8 @@ def _health_payload(state: StateSnapshot, now: float) -> dict:
         "snap_error": state.snap_error,
         "panda_error": state.panda_error,
         "snap_reinit": reinit,
+        "snap_fpga_state": snap_fpga_state,
+        "snap_fpga_last_probe_unix": state.snap_fpga_last_probe_unix,
         "run_tag": state.run_tag,
         "run_started_at_unix": state.run_started_at_unix,
         "run_age_s": run_age_s,

--- a/tests/test_live_status_aggregator.py
+++ b/tests/test_live_status_aggregator.py
@@ -835,3 +835,102 @@ def test_thresholds_from_constructor_override():
         assert agg.thresholds.bands["adc.rms"]["healthy"] == [1.0, 2.0]
     finally:
         agg.stop()
+
+
+# ---------------------------------------------------------------------
+# SNAP FPGA probe
+# ---------------------------------------------------------------------
+
+
+def test_snap_fpga_probe_skipped_when_corr_fresh(agg, seeded, monkeypatch):
+    """When the corr stream is fresh, the probe is a no-op — the corr
+    flow itself proves SNAP liveness, so we don't open extra sockets."""
+    snap, _panda = seeded
+    # Seed a fresh corr_last_unix in state by directly stamping it.
+    agg.state.corr_last_unix = time.time()
+    agg.state.corr_header = {"integration_time": 0.27}
+
+    calls = []
+
+    def fake_probe(*args, **kwargs):
+        calls.append((args, kwargs))
+        return True
+
+    monkeypatch.setattr(
+        "eigsep_observing.live_status.aggregator.probe_snap_fpga",
+        fake_probe,
+    )
+    agg._snap_tick()
+    assert calls == []  # probe must NOT have run
+    # snap_fpga_reachable stays None (probe never ran on this tick).
+    assert agg.state.snap_fpga_reachable is None
+
+
+def test_snap_fpga_probe_runs_when_corr_stale(agg, monkeypatch):
+    """No corr data ever seen → probe should run and update state."""
+    agg.state.corr_config = {"snap_ip": "10.10.10.12"}
+    monkeypatch.setattr(
+        "eigsep_observing.live_status.aggregator.probe_snap_fpga",
+        lambda host, **kw: host == "10.10.10.12",
+    )
+    agg._snap_tick()
+    assert agg.state.snap_fpga_reachable is True
+    assert agg.state.snap_fpga_last_probe_unix is not None
+
+
+def test_snap_fpga_probe_records_unreachable(agg, monkeypatch):
+    agg.state.corr_config = {"snap_ip": "10.10.10.12"}
+    monkeypatch.setattr(
+        "eigsep_observing.live_status.aggregator.probe_snap_fpga",
+        lambda host, **kw: False,
+    )
+    agg._snap_tick()
+    assert agg.state.snap_fpga_reachable is False
+
+
+def test_snap_fpga_probe_gated_by_interval(agg, monkeypatch):
+    """Two ticks within the probe interval must only run the probe
+    once — the inner tick cadence is 0.01 s, the probe interval is
+    5 s by default."""
+    agg.state.corr_config = {"snap_ip": "10.10.10.12"}
+    calls = []
+    monkeypatch.setattr(
+        "eigsep_observing.live_status.aggregator.probe_snap_fpga",
+        lambda host, **kw: calls.append(host) or True,
+    )
+    agg._snap_tick()
+    agg._snap_tick()
+    assert calls == ["10.10.10.12"]
+
+
+def test_snap_fpga_probe_uses_explicit_host_when_no_config(agg, monkeypatch):
+    """If the corr config hasn't been read yet but a snap_fpga_host
+    was passed to __init__, use it."""
+    agg._snap_fpga_host_override = "10.10.10.12"
+    agg.state.corr_config = None
+    monkeypatch.setattr(
+        "eigsep_observing.live_status.aggregator.probe_snap_fpga",
+        lambda host, **kw: host == "10.10.10.12",
+    )
+    agg._snap_tick()
+    assert agg.state.snap_fpga_reachable is True
+
+
+def test_snap_fpga_probe_unknown_when_no_host(agg, monkeypatch):
+    """No snap_ip anywhere → probe skipped, state stays None."""
+    agg.state.corr_config = None
+    agg._snap_fpga_host_override = None
+    calls = []
+    monkeypatch.setattr(
+        "eigsep_observing.live_status.aggregator.probe_snap_fpga",
+        lambda *a, **kw: calls.append(a) or True,
+    )
+    agg._snap_tick()
+    assert calls == []
+    assert agg.state.snap_fpga_reachable is None
+
+
+def test_snapshot_exposes_snap_fpga_fields(agg):
+    snap = agg.snapshot()
+    assert hasattr(snap, "snap_fpga_reachable")
+    assert hasattr(snap, "snap_fpga_last_probe_unix")

--- a/tests/test_live_status_app.py
+++ b/tests/test_live_status_app.py
@@ -328,6 +328,47 @@ def test_health_route_surfaces_snap_reinit_count(agg_primed):
     assert data["snap_reinit"]["seconds_since_reinit"] is not None
 
 
+def test_health_route_snap_fpga_state_live_when_corr_fresh(agg_primed):
+    """corr fresh → state is 'live' regardless of probe result."""
+    agg_primed.state.corr_last_unix = time.time()
+    agg_primed.state.corr_header = {"integration_time": 0.27}
+    agg_primed.state.snap_fpga_reachable = False  # ignored when live
+    client = create_app(agg_primed).test_client()
+    body = client.get("/api/health").get_json()
+    assert body["data"]["snap_fpga_state"] == "live"
+
+
+def test_health_route_snap_fpga_state_reachable_when_probe_ok(
+    agg_primed,
+):
+    """corr stale + probe True → 'reachable'."""
+    agg_primed.state.corr_last_unix = None
+    agg_primed.state.snap_fpga_reachable = True
+    client = create_app(agg_primed).test_client()
+    body = client.get("/api/health").get_json()
+    assert body["data"]["snap_fpga_state"] == "reachable"
+
+
+def test_health_route_snap_fpga_state_unreachable_when_probe_fail(
+    agg_primed,
+):
+    """corr stale + probe False → 'unreachable'."""
+    agg_primed.state.corr_last_unix = None
+    agg_primed.state.snap_fpga_reachable = False
+    client = create_app(agg_primed).test_client()
+    body = client.get("/api/health").get_json()
+    assert body["data"]["snap_fpga_state"] == "unreachable"
+
+
+def test_health_route_snap_fpga_state_unknown_when_no_probe(agg_primed):
+    """corr stale + probe never ran → 'unknown'."""
+    agg_primed.state.corr_last_unix = None
+    agg_primed.state.snap_fpga_reachable = None
+    client = create_app(agg_primed).test_client()
+    body = client.get("/api/health").get_json()
+    assert body["data"]["snap_fpga_state"] == "unknown"
+
+
 def test_corr_route_has_pairs_and_freqs(client):
     body = client.get("/api/corr").get_json()
     data = body["data"]

